### PR TITLE
Upgrade to Swift 5

### PR DESCRIPTION
**Pull request by jessesquires**
- [x] Upgrade to Xcode 10
- [x] Migrate to Swift 5
- [x] Run pod install
- [x] Upgrade `DifferenceKit`
- [x] Bump version numbers
- [x] Update changelog
- [x] Re-gen docs

### CODE REVIEW

**Comment by jessesquires** [comment]
We are currently blocked by https://github.com/ra1028/DifferenceKit/pull/55

**Comment by PlanBot-iOS** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,7 +7,11 @@ The changelog for `ReactiveLists`. Also see the [releases](https://github.com/pl
 NEXT
 ----
 
+0.5.0
+-----
+
 ### Changed
+
```
Comment (asmallteapot) [inline CHANGELOG.md]: Be sure to mention that we’re bumping the Swift version too

**Comment by jessesquires** [comment]
sweet. differencekit just merged that and release 1.1.0

updating

**Comment by jessesquires** [comment]
travis is going to fail because it doesn't have xcode10.2 yet

shocking

**Pull request merged**